redis.ts

```diff
@@ -2359,11 +2359,31 @@ class RedisImpl implements Redis {
   }
 }
 
-export interface RedisConnectOptions extends RedisConnectionOptions {
```

Comment:
Could you move `RedisConnectionOptionsTCP`, `RedisConnectionOptionsUnix` and `RedisConnectionOptions` to `connection.ts`?